### PR TITLE
feat: design token v2 foundation (radius/sizing/motion, vibrant glass surfaces, account insights)

### PR DESCRIPTION
**Pull request by ftchvs**
## Summary

Design token v2 foundation — no dashboard changes yet. `Radius`/`Sizing`/`MotionTokens` (one reduce-motion gate), `SurfaceRank` + `glassSurface()` vibrancy-first surfaces, `displayBalance()`/`dataText()` type tokens, utilization ramp drops yellow (icon ladder carries severity), template-style MenuBarLabel glyph ladder (color-only dot removed), new tested `AccountDetailInsights` model in PlaidBarCore, drill-in a11y strings reworded for the upcoming side panel.

## Autonomous task IDs

- Task ID(s): N/A — user-directed design-direction session (not a backlog T-task)
- Backlog slice: Design direction implementation, stacked PR 246 of 4 (#246 → #247 → #248 → #249)
- Scope note: Tokens, shared modifiers, Core insights model, and menu bar label only; consuming views land in #247

## Agent coordination

- Builder agent: Claude Code (Fable 5), interactive session with Felipe
- Builder branch/worktree: `feature/design-system-foundation` built in worktree `/Users/ftchvs/Developer/PlaidBar/.claude/worktrees/infallible-driscoll-ad8f1c`
- Reviewer/merge steward: Felipe (human approval); Otto may steward the merge after approval
- Coordination note: Stacked base of the chain; merge first; no other agent should push here

## Changed files

- `Sources/PlaidBar/Theme/DesignTokens.swift`
- `Sources/PlaidBar/Theme/Typography.swift`
- `Sources/PlaidBar/Views/MenuBarLabel.swift`
- `Sources/PlaidBar/Views/SharedModifiers.swift`
- `Sources/PlaidBarCore/Models/DashboardDrillInSurface.swift`
- `Sources/PlaidBarCore/Utilities/AccountDetailInsights.swift`
- `Tests/PlaidBarCoreTests/AccountDetailInsightsTests.swift`
- `Tests/PlaidBarCoreTests/PlaidBarCoreTests.swift`

## Local gates

- [x] `git diff --check` clean across the stack
- [x] `swift build` (all targets) and `swift build -Xswiftc -strict-concurrency=complete -Xswiftc -warnings-as-errors` clean on this branch
- [x] PlaidBarServer target built as part of the full-package build (no server code changed in this PR)
- [x] Full `swift test` on this branch: 374/374 pass
- [x] Secret scan of the diff completed: no credentials, tokens, account IDs, or real balances

## GitHub checks

- [ ] Required GitHub checks are green before merge.
- [ ] `otto-openclaw-merge-gate` is green before merge.
- [ ] Skipped checks are expected and not required for this PR.
- [ ] Any failing, pending, cancelled, missing, or ambiguous required check blocks merge.
- [ ] Claude review auth/token/session/setup-only failures are documented as non-blocking, or there are no such failures.

## Privacy and safety impact

- [x] I used only sandbox or synthetic financial data in tests, screenshots, examples, and docs.
- [x] I did not include real Plaid credentials, access tokens, account IDs, transaction exports, raw balances, or screenshots with real financial data.
- [x] This change preserves the local-first boundary: no hosted backend, telemetry, cloud sync, multi-user account system, or cloud AI over private transaction data.
- [x] User-facing copy remains honest about local storage, Plaid credentials, production approval, and unsupported security properties.

## Reviewer local-first and secret-exposure checklist

- [ ] The diff does not introduce, log, display, persist, or document real Plaid secrets, access tokens, public tokens, item IDs, account IDs, transaction exports, raw balances, or real financial screenshots.
- [ ] New status, diagnostics, error, analytics-like, AI, or logging surfaces expose only readiness metadata or synthetic/demo data, never private financial records or identifiers.
- [ ] Any server, storage, setup, or diagnostics change keeps PlaidBar local-first: localhost-only companion server, local data directory, no hosted backend, telemetry, tracking, cloud sync, multi-user account system, or cloud AI over transaction data.
- [ ] Any optional AI behavior is local-only, off by default, explainable, reversible, and non-blocking when no local model runtime is configured.
- [ ] Any documentation, examples, fixtures, tests, and screenshots use sandbox or synthetic data and do not imply production readiness, notarization, distribution, or security properties that have not been verified.

## UI and accessibility checks

- [x] I checked keyboard access and visible focus for UI changes, or this PR has no UI changes.
- [ ] I spot-checked VoiceOver labels or announcements for UI changes, or this PR has no UI changes.
- [x] I kept balances, risk, utilization, errors, and chart meaning understandable without relying on color alone.
- [x] I updated docs or screenshots if user-facing behavior changed.

## Merge safety

- [ ] Final diff reviewed for secrets, private financial data, generated artifacts, scope creep, and unsafe destructive behavior.
- [ ] Merge is within the scoped PlaidBar approval in `docs/autonomous-roadmap.md`, or Felipe explicitly approved this PR.
- [ ] PR head SHA was rechecked immediately before merge.
- [ ] No other agent is actively mutating this branch/worktree.
- [ ] After merge, completed task IDs will be recorded in the roadmap ledger and backlog checklist.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Comment by ftchvs** [comment]
@codex @claude please review

**Comment by ftchvs** [comment]
Addressed the review notes in 7549f98 (cascaded through the stack):

- **`Sizing.hitTargetMin`** raised 24 → 28 with a rationale comment: 28pt is the macOS pointer-target floor for compact controls; 44pt is the touch-input HIG figure and would break menu-bar density. Footer/fly-out glyph buttons grow accordingly in #247/#249.
- **`GlassSurface.emphasizedFill`** now binds the associated tint directly (`guard case let .emphasized(tint)`) — no second dispatch through `SurfaceRank.fill`.
- **`nativePanelSurface`** doc comment now points new surfaces at the rank-based `glassSurface()` system.
- **Menu bar server-unreachable** state now uses `network.slash` (matching the trust-receipt iconography) so it's distinguishable from stale/login at the same warning tier — thanks, good catch on the overload.

**Declined (with reason):** `@available(*, deprecated)` on `heroBalance()` — CI builds with `-warnings-as-errors` and the legacy detail surfaces still call it, so the annotation would fail the build. The right moment is the dead-tab-tree cleanup PR where those call sites are deleted or migrated.

All branches re-verified: strict-concurrency build clean, 374/374 tests pass at the stack tip.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

**Comment by ftchvs** [comment]
@codex review

**Comment by ftchvs** [comment]
Addressed remaining review note in 70469d5: added `maxCategories: 0` and `maxReviewItems: 0` edge-case tests for the `max(0, n)` guards in `AccountDetailInsights.compute`. (Hit-target floor, emphasized-fill double dispatch, offline glyph, and `nativePanelSurface` cross-reference were already handled in aa0a3cf/15dbf76.)

File: Sources/PlaidBar/Views/MenuBarLabel.swift [hunk 1]
```diff
@@ -28,16 +23,18 @@ struct MenuBarLabel: View {
...
 
-    private var statusTint: Color {
+    private var statusSymbolName: String {
         if appState.error != nil || appState.erroredItemCount > 0 {
-            return SemanticColors.negative
+            return "exclamationmark.octagon"
         }
         if appState.needsLoginItemCount > 0 || appState.isSyncStale {
-            return SemanticColors.warning
+            return "exclamationmark.triangle"
```
[the remark this answers is not in the document]
Comment (ftchvs) [reply Sources/PlaidBar/Views/MenuBarLabel.swift:37]: Fixed in 6932676 — MenuBarLabel now checks `!serverConnected` before the stale/login branch, so `network.slash` wins when offline (isSyncStale is usually also true then). Good catch.

File: Sources/PlaidBarCore/Models/DashboardDrillInSurface.swift [hunk 2]
```diff
@@ -135,14 +135,14 @@ public struct DashboardAccountDrillInPath: Sendable, Equatable {
...
-                pointerHelp: "Collapse details for \(displayName)"
+                accessibilityHint: "Press Return or Space to close the account details panel.",
+                accessibilityActionName: "Close account details",
+                pointerHelp: "Close details for \(displayName)"
             )
         }
 
         return Self(
-            accessibilityHint: "Press Return or Space to open the account drill-in below this row.",
+            accessibilityHint: "Press Return or Space to open the account details panel beside the dashboard.",
```
[the remark this answers is not in the document]
Comment (ftchvs) [reply Sources/PlaidBarCore/Models/DashboardDrillInSurface.swift]: Fixed in 6932676 — reverted this hint to 'below this row' on #246; the 'beside the dashboard' wording now lands with the actual side-panel layout in #247 (commit 4e77443), so each PR stays internally consistent.

File: Sources/PlaidBarCore/Utilities/AccountDetailInsights.swift [hunk 3]
```diff
@@ -0,0 +1,246 @@
...
+            .filter(\.pending)
+            .sorted(by: isPreferredForReview)
+            .map { ReviewItem(transaction: $0, reason: .pending) }
+
+        let pendingIds = Set(pending.map(\.id))
+        let large = transactions
+            .filter { transaction in
+                !transaction.pending &&
+                    !transaction.isIncome &&
+                    transaction.displayAmount >= largeAmountThreshold &&
```
[the remark this answers is not in the document]
Comment (ftchvs) [reply Sources/PlaidBarCore/Utilities/AccountDetailInsights.swift:214]: Fixed in 6932676 — added `!transaction.isTransfer` to the large-amount review filter (matching the totals/category paths) plus a regression test. A large transfer no longer shows as spending to review.

**Comment by ftchvs** [comment]
Addressed all three Codex P2 findings in 6932676 (cascaded through the stack):

1. **Offline before stale** — `MenuBarLabel.statusSymbolName` now checks `!serverConnected` before the stale/login branch, so `network.slash` wins when disconnected. Correct call: `isSyncStale` is almost always true offline, which made the new glyph unreachable.
2. **Transfer review exclusion** — added `!transaction.isTransfer` to the large-amount review filter so large card payments / account moves no longer show as spending to review, matching the totals and category paths. Regression test added.
3. **Drill-in copy alignment** — reverted the a11y hint to 'below this row' on this PR and moved the 'beside the dashboard' wording to #247 (commit 4e77443), where the inline panel is actually replaced by the left fly-out. Each PR is now internally consistent.

Strict-concurrency build clean, 377/377 tests pass at the stack tip. Replies posted and threads resolved.

🤖 Generated with [Claude Code](https://claude.com/claude-code)
